driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/syncadapter/SyncListCollectionNamesIterable.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.reactivestreams.client.syncadapter;
+
+import com.mongodb.client.ListCollectionNamesIterable;
+import com.mongodb.lang.Nullable;
+import com.mongodb.reactivestreams.client.ListCollectionNamesPublisher;
+import org.bson.BsonValue;
+import org.bson.conversions.Bson;
+
+import java.util.concurrent.TimeUnit;
+
+final class SyncListCollectionNamesIterable extends SyncMongoIterable<String> implements ListCollectionNamesIterable {
```

Comment:
Is based on `com.mongodb.reactivestreams.client.syncadapter.SyncListCollectionsIterable`.